react/features/youtube-player/middleware.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+
+import { getCurrentConference } from '../base/conference';
+import {
+    PARTICIPANT_LEFT,
+    participantJoined,
+    pinParticipant,
+    participantLeft,
+    getLocalParticipant } from '../base/participants';
+import { MiddlewareRegistry, StateListenerRegistry } from '../base/redux';
+
+import { TOGGLE_SHARED_VIDEO, SET_SHARED_VIDEO_STATUS } from './actionTypes';
+import { setSharedVideoStatus, showEnterVideoLinkPrompt, setSharedVideoOwner } from './actions';
+
+const SHARED_VIDEO = 'shared-video';
+
+/**
+ * Middleware that captures actions related to YouTube video sharing and updates
+ * components not hooked into redux.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+// eslint-disable-next-line no-unused-vars
+MiddlewareRegistry.register(store => next => action => {
+    const state = store.getState();
+    const { ownerId } = state['features/youtube-player'];
+    const conference = getCurrentConference(state);
+    const status = action.status;
+    const localParticipant = getLocalParticipant(state);
+    const localParticipantId = localParticipant && localParticipant.id;
+
+    switch (action.type) {
+    case TOGGLE_SHARED_VIDEO:
+        _toggleSharedVideo(store, next, action);
+        break;
+    case PARTICIPANT_LEFT:
+        if (action.participant.id === ownerId) {
+            store.dispatch(setSharedVideoStatus('stop', ''));
+            const fakeParticipant = state['features/base/participants'].filter(p => p.isFakeParticipant)[0];
+
+            store.dispatch(participantLeft(fakeParticipant.id, conference));
```

Comment:
This feels weird. We should only remove the fake participant which matched the participant who left.